devito/core/gpu.py

```diff
@@ -176,6 +178,9 @@ def _specialize_iet(cls, graph, **kwargs):
         if options['mpi']:
             mpiize(graph, mode=options['mpi'])
 
+        # Lower IncrDimensions so that blocks of arbitrary shape may be used
+        finalize_loop_bounds(graph)
```

Comment:
why should you add this pass when there's no blocking in `_specialize_clusters`. Remember that when the user writes `Operator(..., opt='blocking')` he/she doesn't need to also specify "relax_incr_dimensions". Why? Find out

Reply:
Right, I think it is a leftover from previous gpu blocking attempts